src/components/menu/menu.types.ts

```diff
@@ -75,4 +77,21 @@ export interface MenuItem<T = any> {
      * Value of the menu item.
      */
     value?: T;
+
+    /**
+     * A way of defining a sub-menu for an item.
+     * Set it to an array of `MenuItem`s
+     */
+    items?: Array<MenuItem<T> | ListSeparator>;
```

Comment:
It seems like the change in type of this property from `Array<MenuItem<T> | ListSeparator>` to `Array<MenuItem<T> | ListSeparator> | MenuLoader` (in the next PR) might actually be a breaking change* if we don't release those PRs together, so I'm holding off on merging this for a little bit anyway.

*) Evidence: see the change I had to make to the "open sub-menu programmatically"-example in the next PR: https://github.com/Lundalogik/lime-elements/pull/2577/files#diff-2add54a9e21a898b280c548832b0213ea57812aa63e2a29aecabbc0afd6389e8R87

<img width="781" alt="image" src="https://github.com/Lundalogik/lime-elements/assets/154725/e037d655-302f-4025-b5d8-116614d8896e">